### PR TITLE
ENH: Added sample_dict to `CARDAlleleAnnotationDirectoryFormat` , `CARDGeneAnnotationDirectoryFormat` and `CARDAnnotationDirectoryFormat`

### PR DESCRIPTION
**Pull request by VinzentRisch**
- Added sample_dict function to `CARDAlleleAnnotationDirectoryFormat` , `CARDGeneAnnotationDirectoryFormat` and `CARDAnnotationDirectoryFormat`
- This function returns a dict where keys are sample names and values are paths to the annotation files.

### CODE REVIEW

File: q2_amr/types/_format.py [hunk 1]
```diff
@@ -431,6 +460,13 @@ class CARDGeneAnnotationDirectoryFormat(MultiDirValidationMixin, model.Directory
     def gene_path_maker(self, sample_id):
         return "%s/gene_mapping_data.txt" % sample_id
 
+    def sample_dict(self):
+        sample_dict = {}
+        for sample in self.path.iterdir():
+            file = list(os.path.join(sample, "gene_mapping_data.txt"))
```
Comment (misialq) [inline q2_amr/types/_format.py:466]: Let's not introduce variables where not necessary - you can also just assign the value directly below.

File: q2_amr/types/tests/test_types_formats_transformers.py [hunk 2]
```diff
@@ -315,28 +315,78 @@ def test_CARDAnnotationDirectoryFormat_to_qiime2_Metadata_transformer(self):
...
+
+        obs = annotations.sample_dict()
+        exp = {
+            sample: [
+                os.path.join(dirpath, sample, file)
+                for file in [
+                    "allele_mapping_data.txt",
+                    "overall_mapping_stats.txt",
+                    "sorted.length_100.bam",
+                ]
```
Comment (misialq) [inline q2_amr/types/tests/test_types_formats_transformers.py:373]: I love this loop obsession 😵‍💫

File: q2_amr/types/tests/test_types_formats_transformers.py [hunk 3]
```diff
@@ -315,28 +315,78 @@ def test_CARDAnnotationDirectoryFormat_to_qiime2_Metadata_transformer(self):
...
+        self.assertEqual(obs, exp)
+
+    def test_card_gene_annotation_directory_format_sample_dict(self):
+        dirpath = self.get_data_path("card_gene_annotation")
+        annotations = CARDGeneAnnotationDirectoryFormat(dirpath, mode="r")
+
+        obs = annotations.sample_dict()
+        exp = {
+            "sample1": list(os.path.join(dirpath, "sample1", "gene_mapping_data.txt")),
+            "sample2": list(os.path.join(dirpath, "sample2", "gene_mapping_data.txt")),
```
Comment (misialq) [inline q2_amr/types/tests/test_types_formats_transformers.py:386]:
```suggestion
            "sample1": [os.path.join(dirpath, "sample1", "gene_mapping_data.txt"),],
            "sample2": [os.path.join(dirpath, "sample2", "gene_mapping_data.txt"),],
```

**Review by misialq — CHANGES_REQUESTED** [review]
Hey @VinzentRisch, looks good! See some small suggestions below, otherwise seems alright 😄

**Comment by VinzentRisch** [comment]
Hi @misialq  
Sorry for all the loops, will avoid this in the future :)

**Review by misialq — APPROVED** [review]
Awesome, thanks @VinzentRisch! Feel free to merge whenever ready :)

**Pull request merged**